tests/orchestrations_test.go

```diff
@@ -715,6 +715,142 @@ func Test_TerminateOrchestration(t *testing.T) {
 	)
 }
 
+func Test_TerminateOrchestration_Recursive(t *testing.T) {
```

Comment:
Could you add a test with 2 Terminate (and one with 2 Purge) ops in parallel to confirm there are no concurrency issues (such as transactions or race conditions)?

Reply:
Do you mean to add 2 Terminate Ops to same instance of workflow in parallel? And each of them should return success? For terminate I would need to check if that's fine
For purge, that would fail right? As once the one of them succeeds, other one would return with `ErrInstanceNotFound`.